packages/legend-extension-dsl-serializer/src/models/protocols/pure/DSLSerializer_PureProtocolProcessorPlugin.ts

```diff
@@ -170,11 +175,22 @@ export class DSLSerializer_PureProtocolProcessorPlugin extends DSLMapping_PurePr
             elementProtocol.name,
           );
           const element = getSchemaSet(path, context.graph);
-          element.format = guaranteeNonEmptyString(
-            elementProtocol.format,
-            `Schema set 'format' field is missing or empty`,
-          );
-          element.schemas = elementProtocol.schemas;
+          element.format =
+            Object.values(FORMAT_TYPE).find(
```

Comment:
You would get an error message from engine side if we don't support the format type. Do you want me to include the condition here as well?

Reply:
> I think this is not proper, since we are building the graph from the input, if the input contains a format type that we don't support, we should throw.

Compilation fails and you would get an error message from engine side if we don't support the format type. Do you want me to include the condition here as well?

Reply:
yes, I think that's fair to throw if they give us a messed up protocol model